kalite/coachreports/api_views.py

```diff
@@ -1,382 +1,87 @@
-from collections_local_copy import OrderedDict
+from math import ceil
 
-from django.conf import settings; logging = settings.LOG
-from django.core.exceptions import PermissionDenied, ValidationError
-from django.http import HttpResponseNotFound, Http404
-from django.shortcuts import get_object_or_404
-from django.views.decorators.csrf import csrf_exempt
 from django.utils.translation import ugettext as _
+from django.db.models import Q
 
-from .forms import DataForm
-from fle_utils.internet.classes import JsonResponse
-from fle_utils.internet.decorators import api_handle_error_with_json
-from fle_utils.testing.decorators import allow_api_profiling
-from kalite.facility.models import Facility, FacilityUser, FacilityGroup
-from kalite.main.models import VideoLog, ExerciseLog, UserLog, UserLogSummary
-from kalite.topic_tools import get_topic_by_path, get_node_cache, get_exercise_cache
-from kalite.i18n import lcode_to_django_lang
+from fle_utils.internet.classes import JsonResponse, JsonResponseMessage, JsonResponseMessageError
 
+from kalite.main.models import ExerciseLog, VideoLog, ContentLog
+from kalite.facility.models import FacilityUser
+from kalite.shared.decorators.auth import require_admin
+from kalite.topic_tools import get_topic_leaves, get_exercise_cache, get_content_cache
 
-# Global variable of all the known stats, their internal and external names,
-#    and their "datatype" (which is a value that Google Visualizations uses)
-stats_dict = [
-    {"key": "pct_mastery",        "name": _("Mastery"),          "type": "number", "description": _("Percent of exercises mastered (at least 10 consecutive correct answers)"), "timeline": True},
-    {"key": "effort",             "name": _("Effort"),           "type": "number", "description": _("Combination of attempts on exercises and videos watched.")},
-    {"key": "ex:attempts",        "name": _("Attempts"),   "type": "number", "description": _("Number of times submitting an answer to an exercise.")},
-    {"key": "ex:streak_progress", "name": _("Streak"),     "type": "number", "description": _("Maximum number of consecutive correct answers on an exercise.")},
-    {"key": "ex:points",          "name": _("Exercise points"),    "type": "number", "description": _("[Pointless at the moment; tracks mastery linearly]")},
-    { "key": "ex:completion_timestamp", "name": _("Time exercise completed"),"type": "datetime", "description": _("Day/time the exercise was completed.") },
-    {"key": "vid:points",          "name": _("Video points"),      "type": "number", "description": _("Points earned while watching a video (750 max / video).")},
-    { "key": "vid:total_seconds_watched","name": _("Video time"),   "type": "number", "description": _("Total seconds spent watching a video.") },
-    { "key": "vid:completion_timestamp", "name": _("Time video completed"),"type": "datetime", "description": _("Day/time the video was completed.") },
-]
+@require_admin
+def learner_logs(request):
```

Comment:
Tests would be good! I think you can use the ClientTestCase here without having to use all the selenium machinery.